src/utils.ts

```diff
@@ -25,3 +32,93 @@ export function toAbsoluteUrl(host: string | undefined, path: string): string {
     return host + '/' + path;
   }
 }
+
+/*
+Type with the possible ways to define a color in maplibre
+*/
+export type mbColorDefinition =
+  | PropertyValueSpecification<ColorSpecification>
+  | DataDrivenPropertyValueSpecification<ColorSpecification>;
+
+/*
+A selection of the different maplibre-gl-js properties that refer to a color.
+Taken from maplibre https://github.com/maplibre/maplibre-gl-js/blob/main/src/style-spec/types.g.ts
+*/
+export type layerPaintProperty = {
+  'background-color'?: PropertyValueSpecification<ColorSpecification>;
+  'circle-color'?: DataDrivenPropertyValueSpecification<ColorSpecification>;
+  'circle-stroke-color'?: DataDrivenPropertyValueSpecification<ColorSpecification>;
+  'fill-color'?: DataDrivenPropertyValueSpecification<ColorSpecification>;
+  'fill-extrusion-color'?: DataDrivenPropertyValueSpecification<ColorSpecification>;
+  'fill-outline-color'?: DataDrivenPropertyValueSpecification<ColorSpecification>;
+  'icon-color'?: DataDrivenPropertyValueSpecification<ColorSpecification>;
+  'icon-halo-color'?: DataDrivenPropertyValueSpecification<ColorSpecification>;
+  'line-color'?: DataDrivenPropertyValueSpecification<ColorSpecification>;
+  'text-color'?: DataDrivenPropertyValueSpecification<ColorSpecification>;
+  'text-halo-color'?: DataDrivenPropertyValueSpecification<ColorSpecification>;
+};
+
+export type blendMode =
+  | 'multiply'
+  | 'darken'
+  | 'lighten'
+  | 'screen'
+  | 'overlay'
+  | 'burn'
+  | 'dodge'
+  | 'mix';
+
+/*
+Function to transform a maplibre color definition by a given function.
+*/
+function transformColor(paintColor: mbColorDefinition, func: Function): mbColorDefinition {
+  if (typeof paintColor == 'string') {
+    const modifiedColor = func(paintColor);
+    return `rgba(${modifiedColor.rgba().join(',')})`;
+  } else if (
+    typeof paintColor == 'object' &&
+    'stops' in paintColor &&
+    Array.isArray(paintColor?.stops)
+  ) {
+    const stops = paintColor['stops'].map((stop) => {
+      const newColor = transformColor(stop[1], func);
+      return [stop[0], newColor];
+    });
+    const newPaintColor = Object.assign({}, paintColor, { stops: stops });
```

Comment:
nit, untested suggestion 

I think you can also use [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_object_literals). 

```suggestion
    const newPaintColor = { ...paintColor, ...stops };
```